packages/commonwealth/server/farcaster/frames/contest/viewLeaderboard.tsx

```diff
@@ -1,14 +1,105 @@
+import { error } from 'frames.js/core';
 import { Button } from 'frames.js/express';
 import React from 'react';
 import { frames } from '../../config';
-import { CardWithText } from '../../utils';
+import { getInvertedColor, getLeaderboard, getRandomColor } from '../../utils';
+
+const randomColor = getRandomColor();
+const invertedColor = getInvertedColor(randomColor);
+
+export const viewLeaderboard = frames(async (ctx) => {
+  const fromMain = ctx.searchParams.fromMain;
+
+  let leaderboardEntries;
+
+  try {
+    leaderboardEntries = await getLeaderboard();
+  } catch (err) {
+    console.log('error', err);
+    return error('Something went wrong');
+  }
+
+  const noEntries = leaderboardEntries.length === 0;
+  const entry = leaderboardEntries[0];
 
-export const viewLeaderboard = frames(async () => {
   return {
-    image: CardWithText({ text: 'Leaderboard' }),
+    image: (
+      <div
+        style={{
+          backgroundColor: '#2A2432',
+          color: 'white',
+          padding: '40px',
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          width: '100%',
+          height: '100%',
+          textAlign: 'center',
+        }}
+      >
+        {noEntries ? (
+          <p>No entries yet</p>
+        ) : (
+          <div
+            style={{
+              display: 'flex',
+              flexDirection: 'column',
+              alignItems: 'center',
+            }}
+          >
+            <div
+              style={{
+                borderRadius: '50%',
+                width: '100px',
+                height: '100px',
+                backgroundColor: `#${randomColor}`,
+                color: `#${invertedColor}`,
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+                fontSize: '56px',
+              }}
+            >
+              {entry.nickname.charAt(0).toUpperCase()}
```

Comment:
if the intent is to capitalize, can we `_.capitalize`?

Reply:
wouldn't worry about it because it is just a mock for testing purpose, it will be deleted soon